validator/sawtooth_validator/execution/context_manager.py

```diff
@@ -210,112 +272,120 @@ def commit_context(self, context_id_list, virtual):
 
     def delete_context(self, context_id_list):
-        """
-        Part of the interface to the Executor.
-        Throws away contexts, eg. InvalidTransaction
+        """Delete contexts from the ContextManager.
+
         Args:
             context_id_list (list): a list of context ids
 
         Returns:
-            void
+            None
 
         """
         for c_id in context_id_list:
-            with self._shared_lock:
-                if c_id in self._contexts:
-                    del self._contexts[c_id]
+            if c_id in self._contexts:
+                del self._contexts[c_id]
 
     def get(self, context_id, address_list):
-        """
+        """Get the values associated with list of addresses, for a specific
+        context referenced by context_id.
 
         Args:
-            context_id (str): the return value of create_context
+            context_id (str): the return value of create_context, referencing
+                a particular context.
             address_list (list): a list of address strs
 
         Returns:
             values_list (list): a list of (address, value) tuples
         """
-        with self._shared_lock:
-            if context_id not in self._contexts:
-                return []
-        with self._shared_lock:
-            context = self._contexts.get(context_id)
+
+        if context_id not in self._contexts:
+            return []
+        context = self._contexts.get(context_id)
         return [(a, f.result())
                 for a, f in context.get_from_prefetched(address_list)]
 
     def set(self, context_id, address_value_list):
-        """
-        speculatively sets addresses to a value,
-        can be destroyed or committed to the merkle store
+        """Within a context, sets addresses to a value.
+
         Args:
             context_id (str): the context id returned by create_context
             address_value_list (list): list of {address: value} dicts
 
-        Returns (boolean): True, or False whether the
-
+        Returns:
+            (bool): True if the operation is successful, False if
+                the context_id doesn't reference a known context.
         """
-        with self._shared_lock:
-            if context_id not in self._contexts:
-                LOGGER.info("Context_id not in contexts, %s", context_id)
-                return False
-        with self._shared_lock:
-            context = self._contexts.get(context_id)
-            context.can_set(address_value_list)
-            add_value_dict = {}
-            for d in address_value_list:
-                for add, val in d.items():
-                    add_value_dict[add] = val
-            context.set_futures(add_value_dict)
+
+        if context_id not in self._contexts:
+            LOGGER.warning("Context_id not in contexts, %s", context_id)
+            return False
+
+        context = self._contexts.get(context_id)
+        context.can_set(address_value_list)
```

Comment:
A comment on this can_set is probably needed. It's unclear that this is intended to raise an exception.